app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -61,48 +61,30 @@ class SchoolDisciplineDashboard extends React.Component {
       title: "Incidents by " + selectedChart};
   }
 
-  sortChartData(chart) {
-    switch(chart.type) {
-      case 'time': return this.sortByTime(chart.disciplineIncidents);
-      case 'day': return this.sortByDay(chart.disciplineIncidents);
-      default: return chart.disciplineIncidents;
+  sortChartKeys(chartKeys) {
+    switch(this.state.selectedChart) {
+    case 'time': return this.sortedTimes(chartKeys);
+    case 'day': return this.sortedDays(chartKeys);
+    case 'grade': return this.sortedGrades(chartKeys);
+    default: return chartKeys;
     }
   }
 
-  sortByDay(incidentsGroupedByDay) {
-    const dayOrder = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
-    let sortedHash = {};
-    dayOrder.forEach((day) => {
-      if (incidentsGroupedByDay[day]) sortedHash[day] = incidentsGroupedByDay[day];
-    });
-    return sortedHash;
+  sortedDays(chartKeys) {
+    return ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'].filter((day) => chartKeys.indexOf(day) >= 0);
```

Comment:
I'm not sure filtering out days with zero discipline incidents is what we want for the x-axis. If I'm looking at a days of the week chart, I'd expect to see MTWThFSSun on the x-axis, even if there are (for example) no incidents on Tuesday, Wednesday, or Sunday. In the context of this chart I think days with no incidents could be especially interesting . . . "why are there no incidents on Mondays but lots on Fridays?," for example.